gui/src/main.rs

```diff
@@ -376,90 +412,79 @@ impl App for Gui {
                                 });
                             })
                             .body(|mut body| {
-                                for env_var in self.env_vars.borrow_mut().iter_mut() {
-                                    let (enabled, key, value, r#type) = env_var;
-                                    body.row(30.0, |mut row| {
-                                        row.col(|ui| {
-                                            ui.checkbox(enabled, "");
-                                        });
-                                        row.col(|ui| {
-                                            ui.text_edit_singleline(key);
-                                        });
-                                        row.col(|ui| {
-                                            ui.text_edit_singleline(value);
-                                        });
-                                        row.col(|ui| {
-                                            ui.text_edit_singleline(r#type);
-                                        });
-                                    });
+                                let selected_environment =
+                                    self.selected_environment.borrow_mut().clone();
+                                if let Some(env) = selected_environment {
+                                    if let Some(values) = env.values {
+                                        for mut env_var in values {
+                                            body.row(30.0, |mut row| {
+                                                row.col(|ui| {
+                                                    ui.checkbox(&mut env_var.enabled, "");
+                                                });
+                                                row.col(|ui| {
+                                                    ui.text_edit_singleline(&mut env_var.key);
+                                                });
+                                                row.col(|ui| {
+                                                    ui.text_edit_singleline(&mut env_var.value);
+                                                });
+                                                row.col(|ui| {
+                                                    ui.text_edit_singleline(&mut env_var.r#type);
+                                                });
+                                            });
+                                        }
+                                    }
                                 }
-                                //TODO - make inputs work with a struct. can only get things to
```

Comment:
Got things working as a struct now that state is in its own property. Removed this commented out block.